### PR TITLE
[T13-4] CAPTracker

### PR DESCRIPTION
**Pull request by jeremiah-ang**
@alexkmj 
@josephambe
@jeremyyew 
@BugEyedBug

Pull request for [T13-4] CAPTracker

### CODE REVIEW

File: docs/AboutUs.adoc [hunk 1]
```diff
@@ -4,53 +4,52 @@
...
+CAPTracker is for those students who prefer to use a desktop app for calculating and managing their CAP. More importantly CAPTracker is optimized for those who prefer to work with a Command Line Interface (CLI) while still having the benefits of a Graphical User Interface (GUI). If you can type fast, CAPTracker is the ideal application to calculate your current CAP, and predict what grades you need in modules you haven’t taken to achieve your ideal CAP. +
 We are a team based in the http://www.comp.nus.edu.sg[School of Computing, National University of Singapore].
 
 == Project Team
 
-=== John Doe
-image::damithc.jpg[width="150", align="left"]
-{empty}[http://www.comp.nus.edu.sg/~damithch[homepage]] [https://github.com/damithc[github]] [<<johndoe#, portfolio>>]
+=== Alex Koh
+image::alexkmj.jpg[width="150", align="left"]
```
Comment (parvathy-sudhir-pillai) [inline docs/AboutUs.adoc]: Images except Jeremiah's not visible. 
The portfolio links of other developers give a 404 error.

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -12,19 +12,19 @@ ifdef::env-github[]
...
+:repoURL: https://github.com/CS2103-AY1819S1-T13-4/main/tree/master
 
-By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
+By: `T13-4`      Since: `Aug 2018`      Licence: `MIT`
 
 == Introduction
 
-AddressBook Level 4 (AB4) is for those who *prefer to use a desktop app for managing contacts*. More importantly, AB4 is *optimized for those who prefer to work with a Command Line Interface* (CLI) while still having the benefits of a Graphical User Interface (GUI). If you can type fast, AB4 can get your contact management tasks done faster than traditional GUI apps. Interested? Jump to the <<Quick Start>> to get started. Enjoy!
+CAPTracker is for those students who prefer to use a desktop app for calculating and managing their CAP. More importantly CAPTracker is *optimized for those who prefer to work with a Command Line Interface* (CLI) while still having the benefits of a Graphical User Interface (GUI). If you can type fast, CAPTracker is the ideal application to calculate your current CAP, and predict what grades you need in modules you haven’t taken to achieve your ideal CAP. Interested? Jump to the <<Quick Start>> to get started. Enjoy!
```
Comment (parvathy-sudhir-pillai) [inline docs/UserGuide.adoc]: Good job updating the user guide

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -533,360 +533,97 @@ a. Include those libraries in the repo (this bloats the repo size) +
...
-Use case resumes at step 2.
-
-_{More to be added}_
+=== Use case: Calculate predicted CAP
+  . User enters all modules - modules with grades (previous or target), modules without grades (unknown).
+  . System immediately displays confirmation and current CAP.
+
+=== Use case:  View grades needed to achieve CAP goal (based on previous grades)
+  . Use-Case-B.1.
+  . User enters CAP goal.
```
Comment (parvathy-sudhir-pillai) [inline docs/DeveloperGuide.adoc]: System does not respond after entering the CAP goal?

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -533,360 +533,97 @@ a. Include those libraries in the repo (this bloats the repo size) +
...
+  ... System displays target grades for each module (numeric).
+  .. **V2.0:**
+  ... System displays target grades for each module (numeric and letter).
+  ... User edits target grades.
+  ... System displays new target grades for each module.
+
+=== Use case:  Save entered module
+  . User enters some modules
+  . System saves the modules
+  . User restarts the application
```
Comment (parvathy-sudhir-pillai) [inline docs/DeveloperGuide.adoc]: The system can show the saved modules only after restart?

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -533,360 +533,97 @@ a. Include those libraries in the repo (this bloats the repo size) +
...
-* 3a. The given index is invalid.
-+
-[none]
-** 3a1. AddressBook shows an error message.
-+
-Use case resumes at step 2.
-
-_{More to be added}_
+=== Use case: Calculate predicted CAP
+  . User enters all modules - modules with grades (previous or target), modules without grades (unknown).
```
Comment (parvathy-sudhir-pillai) [inline docs/DeveloperGuide.adoc]: Should be done in a loop. Refer to the notation for writing loops in use cases and update.

File: docs/DeveloperGuide.adoc [hunk 6]
```diff
@@ -533,360 +533,97 @@ a. Include those libraries in the repo (this bloats the repo size) +
...
+  ... System displays target grades for each module (numeric and letter).
+  ... User edits target grades.
+  ... System displays new target grades for each module.
+
+=== Use case:  Save entered module
+  . User enters some modules
+  . System saves the modules
+  . User restarts the application
+  . User list entered modules
+  . System displays saved modules
```
Comment (parvathy-sudhir-pillai) [inline docs/DeveloperGuide.adoc]: Try to follow the use-case notations as given in the example one provided by AddressBook with MSS, Extensions etc.

**Review by parvathy-sudhir-pillai — COMMENTED** [review]
Overall a good effort.
 
Suggestions to improve further:
* All developers to update their respective portions of user and developer guides.
* Start to make changes in java files.

File: README.adoc [hunk 7]
```diff
@@ -15,19 +14,12 @@ ifndef::env-github[]
...
-* It is a Java sample application intended for students learning Software Engineering while using Java as the main programming language.
-* It is *written in OOP fashion*. It provides a *reasonably well-written* code example that is *significantly bigger* (around 6 KLoC)than what students usually write in beginner-level SE modules.
-* What's different from https://github.com/se-edu/addressbook-level3[level 3]:
-** A more sophisticated GUI that includes a list  panel and an in-built Browser.
-** More test cases, including automated GUI testing.
-** Support for _Build Automation_ using Gradle and for _Continuous Integration_ using Travis CI.
+* CAPTracker is for those students who prefer to use a desktop app for calculating and managing their CAP. More importantly CAPTracker is optimized for those who prefer to work with a Command Line Interface (CLI) while still having the benefits of a Graphical User Interface (GUI). If you can type fast, CAPTracker is the ideal application to calculate your current CAP, and predict what grades you need in modules you haven’t taken to achieve your ideal CAP.
```
Comment (parvathy-sudhir-pillai) [inline README.adoc]: "Ideal" CAP seems a bit unrealistic. May be "target" CAP is a better word.

Comment (parvathy-sudhir-pillai) [reply README.adoc]: AB-4 should be part of acknowledgements

File: docs/UserGuide.adoc [hunk 8]
```diff
@@ -46,177 +46,81 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
...
 ****
 
 Examples:
 
-* `edit 1 p/91234567 e/johndoe@example.com` +
-Edits the phone number and email address of the 1st person to be `91234567` and `johndoe@example.com` respectively.
-* `edit 2 n/Betsy Crower t/` +
-Edits the name of the 2nd person to be `Betsy Crower` and clears all existing tags.
+* `edit CS2103 CS1231 2 1` +
+Edits the module code and year/semester of `CS2103` module to be `CS1231` and `2 1` respectively.
```
Comment (parvathy-sudhir-pillai) [inline docs/UserGuide.adoc]: "year/semester" or "year and (or) semester"?

File: docs/AboutUs.adoc (hunk 1 above)
Comment (parvathy-sudhir-pillai) [inline docs/AboutUs.adoc]: Alex's image is still not visible

File: docs/AboutUs.adoc [hunk 9]
```diff
@@ -4,53 +4,52 @@
...
-Responsibilities: Data
+Responsibilities: [Refer to W6.10]
 
 '''
 
-=== Johnny Roe
-image::m133225.jpg[width="150", align="left"]
-{empty}[http://github.com/m133225[github]] [<<johndoe#, portfolio>>]
+=== Jeremiah Ang
+image::jeremiah-ang.png[width="150", align="left"]
```
Comment (parvathy-sudhir-pillai) [inline docs/AboutUs.adoc:40]: Suggest to add a photo where face is not too small to identify you

File: docs/DeveloperGuide.adoc [hunk 10]
```diff
@@ -533,360 +533,98 @@ a. Include those libraries in the repo (this bloats the repo size) +
...
-==== Full Solution
-
-See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-by-step solution.
 
 [appendix]
-== Product Scope
-
-*Target user profile*:
+== Target User Profile
+- A NUS Student who has a need to keep track of current CAP, calculate expected CAP, and grades required to achieve desired CAP.
```
Comment (parvathy-sudhir-pillai) [inline docs/DeveloperGuide.adoc]: 'An' NUS student

File: docs/DeveloperGuide.adoc [hunk 11]
```diff
@@ -533,360 +533,98 @@ a. Include those libraries in the repo (this bloats the repo size) +
...
-|`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
-
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
-
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
-|=======================================================================
-
-_{More to be added}_
+*Must-Have*
+
```
Comment (parvathy-sudhir-pillai) [inline docs/DeveloperGuide.adoc:557]: Suggestion to do a spellcheck and grammar check

File: docs/DeveloperGuide.adoc [hunk 12]
```diff
@@ -533,360 +533,98 @@ a. Include those libraries in the repo (this bloats the repo size) +
...
+*module title*, MCs, grade, semester taken) so that I can calculate
+average mark I need to graduate with desired CAP.
+2.  As a user I can delete modules so if I change my mind or fail a
+module I can re-calculate my average.
+3.  As a user I can delete modules so that if i entered a wrong module
+or failed it, I can remove it.
+4.  As a user I can edit the marks I’ve entered previously so that I can
+update my CAP.
+5.  As a user, I can enter in a CAP goal so I can track my progress in
+my modules.
```
Comment (parvathy-sudhir-pillai) [inline docs/DeveloperGuide.adoc]: Seems too vague. How do you track progress with only grades? What do you track progress of?

File: docs/DeveloperGuide.adoc [hunk 13]
```diff
@@ -533,360 +533,98 @@ a. Include those libraries in the repo (this bloats the repo size) +
...
+6.  As a user, I can enter a CAP goal so that I can keep track of the
+progress of my course
```
Comment (parvathy-sudhir-pillai) [inline docs/DeveloperGuide.adoc:568]: What's the difference between story 5 and 6?

**Review by parvathy-sudhir-pillai — COMMENTED** [review]
Overall:
Good effort on meeting all requirements for the v1.1 team component. 
Your metlife preview is not loading. Check log and correct the issue.
Suggestions below for improvements in the upcoming versions:

* Developer Guide
screen shot 2018-10-07 at 12 24 53 am

Add more use cases

* Other comments inline

File: src/main/java/seedu/address/logic/commands/CapCommand.java [hunk 14]
```diff
@@ -0,0 +1,25 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.logic.CommandHistory;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+
+/**
+ * Show CAP based on existing modules.
```
Comment (parvathy-sudhir-pillai) [inline src/main/java/seedu/address/logic/commands/CapCommand.java]: "Shows" (descriptive style)

File: src/main/java/seedu/address/logic/parser/GoalCommandParser.java [hunk 15]
```diff
@@ -0,0 +1,31 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
```
Comment (parvathy-sudhir-pillai) [inline src/main/java/seedu/address/logic/parser/GoalCommandParser.java:4]: Can avoid unnecessary line breaks between import statements

File: src/main/java/seedu/address/logic/parser/GoalCommandParser.java [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+import seedu.address.logic.commands.GoalCommand;
+
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parse User Input
```
Comment (parvathy-sudhir-pillai) [inline src/main/java/seedu/address/logic/parser/GoalCommandParser.java]: "Parses"

File: docs/AboutUs.adoc (hunk 1 above)
Comment (parvathy-sudhir-pillai) [reply docs/AboutUs.adoc]: Also, use pictures which show your face clearly @jeremiah-ang

**Review by parvathy-sudhir-pillai — COMMENTED** [review]
### Feedback for v1.2

**User Guide**
* This command seems to be doing something different from the description. The command heading is compute CAP, whereas what it does is to set target CAP
t13-1

**Developer Guide**
* No significant updates since v1.2
* Not enough new/updated design descriptions and UML diagrams to give feedback on. Please try to add more soon.
* While drawing SDs, keep in mind that we discourage low level diagrams that span multiple components. Try to draw diagrams at multiple levels: one that shows component level interactions and others shown low level interactions within one components.

### Overall
Work on making the product user-testable soon, and catch up on documentation work. The progress at the moment seems a bit behind, at least w.r.t Documentation, than what is required at this stage 😟

**Comment by alexkmj** [comment]
Close this PR and create a new PR to resolve the build fail issue. [Refer here.](https://github.com/nus-cs2103-AY1819S1/forum/issues/118)

**Comment by damithc** [comment]
Closed because Travis seems to be misbehaving

**Pull request closed without merging**